src/evm-block-extractor/src/rpc.rs

```diff
@@ -119,7 +120,16 @@ impl EthServer for EthImpl {
                 jsonrpsee::types::error::ErrorCode::InternalError
             })?;
 
-        Ok(block)
+        if full {
+            serde_json::to_value(block)
+        } else {
+            let converted_block: Block<H256> = block.into();
```

Comment:
We fetch all the data from the DB and then discard part of it? We should use different queries instead if full is true or false

Reply:
Currently we store full blocks in the DB so making those changes would require changing the DB structures everywhere. Also in our other apps we actually don't use send requests to query block with tx hashes, I rather fixed this for consistency.
Do you think optimizing it worth efforts?

Reply:
I think we should persist the TXs in a separate table indexed by hash as we do for the receipts. This would unlock more future scenarios and ETH RPC endpoints

Reply:
@GraDKh the transactions are now in a dedicated table